wallet/walletrpc.c

```diff
@@ -915,6 +923,92 @@ static void json_add_txtypes(struct json_stream *result, const char *fieldname,
 	json_array_end(result);
 }
 
+static void json_transaction_details(struct json_stream *response,
+				     const struct wallet_transaction *tx)
+{
+	struct wally_tx *wtx = tx->tx->wtx;
+
+		json_object_start(response, NULL);
+		json_add_txid(response, "hash", &tx->id);
+		json_add_hex_talarr(response, "rawtx", tx->rawtx);
+		json_add_u64(response, "blockheight", tx->blockheight);
+		json_add_num(response, "txindex", tx->txindex);
+
+		if (tx->annotation.type != 0)
+			json_add_txtypes(response, "type", tx->annotation.type);
+		else
+			json_add_null(response, "type");
+
+		if (tx->annotation.channel.u64 != 0)
+			json_add_short_channel_id(response, "channel", &tx->annotation.channel);
+		else
+			json_add_null(response, "channel");
```

Comment:
I far prefer simply omit fields which don't make sense for a field, rather than set them to null.